### DISCUSSION TITLE
resume_form_checkpoint with different hyperparameters

### DISCUSSION THREAD

**Discussion by sirtris** [Lightning-AI/pytorch-lightning#14144] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi,
I am training a classifier with a pretrained backbone. I froze the backbone and only trained a linear classifier for some epochs. Now I want to continue training with an unfrozen backbone and a smaller learning rate. 

Can this be done by just just using the `resume_from_checkpoint` argument in the trainer or will the encoder still be frozen?
If it is not possible what would the simplest way to achieve my goal be?

Thanks

**Comment by rohitgr7** [comment, marked as answer]
> Can this be done by just just using the resume_from_checkpoint argument in the trainer or will the encoder still be frozen?

no, since Trainer doesn't know what your backbone is and it doesn't freeze any module manually.
you can check out [BackboneFinetuning](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.BackboneFinetuning.html) for your usecase.

**Reply by sirtris** [reply]
k, THX.

So the checkpoint does not save the `requires_grad` parameter of a layer?

**Reply by rohitgr7** [reply]
it just saves what `model.state_dict` returns which comes from pytorch and AFAIK it doesn't. But you can verify

**Reply by sirtris** [reply]
Thanks for pointing out the `BackboneFinetuning` callback. It seems to do exactly what I was trying to do.
